pkg/kanctl/profile.go

```diff
@@ -284,13 +303,17 @@ func constructSecret(ctx context.Context, lP *locationParams, cmd *cobra.Command
 		data[secretField] = storageKey
 		secretname = "azure"
 	}
-	return &v1.Secret{
+	secret := &v1.Secret{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      fmt.Sprintf(secretFormat, secretname, randString(6)),
 			Namespace: lP.namespace,
 		},
 		StringData: data,
-	}, nil
+	}
+	if roleKey != "" {
```

Comment:
isn't this specific to S3?

Reply:
Currently we validate the profile to check for `SecretType` when credentialType = "Secret" and role != ""